Buildings/Controls/OBC/ASHRAE/PrimarySystem/BoilerPlant/Generic/PlantEnable.mo

```diff
@@ -0,0 +1,300 @@
+    annotation (Placement(transformation(extent={{-120,40},{-100,60}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Latch lat
+    "Maintain plant status till the conditions to change it are met"
+    annotation (Placement(transformation(extent={{120,-10},{140,10}})));
+
+  Buildings.Controls.OBC.CDL.Logical.MultiAnd mulAnd(final nu=4)
+    "Check if all the conditions for enabling plant have been met"
+    annotation (Placement(transformation(extent={{80,110},{100,130}})));
+
+  Buildings.Controls.OBC.CDL.Logical.MultiOr mulOr(final nu=3)
+    "Check if any conditions except plant-on time have been satisfied to disable plant"
+    annotation (Placement(transformation(extent={{30,-80},{50,-60}})));
+
+  Buildings.Controls.OBC.CDL.Logical.And and2
+    "Check if all conditions have been met to disable the plant"
+    annotation (Placement(transformation(extent={{80,-40},{100,-20}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Not not1
+    "Logical Not"
+    annotation (Placement(transformation(extent={{-10,-120},{10,-100}})));
+
+  Buildings.Controls.OBC.CDL.Continuous.AddParameter addPar(final p=TLocOut,
+                                                            final k=-1)
+    "Compare measured outdoor air temperature to boiler lockout temperature"
+    annotation (Placement(transformation(extent={{-150,-60},{-130,-40}})));
+
+  Buildings.Controls.OBC.CDL.Continuous.Hysteresis hys(final uLow=-locDt,
+                                                       final uHigh=0)
+    "Hysteresis loop to prevent cycling caused by measured value"
+    annotation (Placement(transformation(extent={{-120,-60},{-100,-40}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Not not3
+    "Logical Not"
+    annotation (Placement(transformation(extent={{-70,-40},{-50,-20}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Timer tim1
+    "Time since number of requests was greater than number of ignores"
+    annotation (Placement(transformation(extent={{-40,-40},{-20,-20}})));
+
+  Buildings.Controls.OBC.CDL.Continuous.GreaterThreshold greThr1(final threshold=staOnReqTim)
+    "Time limit for receiving requests to maintain status on"
+    annotation (Placement(transformation(extent={{-10,-40},{10,-20}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Not not4
+    "Logical Not"
+    annotation (Placement(transformation(extent={{-20,60},{0,80}})));
+
+  Buildings.Controls.OBC.CDL.Continuous.GreaterEqualThreshold greEquThr1(final
+      threshold=plaOnThrTim)
+    "Check if minimum amount of time to maintain the plant on has elapsed"
+    annotation (Placement(transformation(extent={{40,0},{60,20}})));
+
+  Buildings.Controls.OBC.CDL.Continuous.GreaterEqualThreshold greEquThr(final
+      threshold=plaOffThrTim)
+    "Check if minimum amount of time to maintain the plant off has elapsed"
+    annotation (Placement(transformation(extent={{40,60},{60,80}})));
+
+  Buildings.Controls.OBC.CDL.Logical.Pre pre1
+    "Feedback delay"
```

Comment:
We usually say `Logical pre block`. Delay often has other meanings.

Reply:
There is an another aspect to this, and that is the pre block is a bit tricky to translate into other control languages. What I've been doing is I've been looking for ways to replace it with some other formulation. In this case one could use a `TrueFalseHold` block, in which one can set the duration of both true and false hold to a parameter. If you were to put this block after the latch you would likely be able to get rid of that entire central bit starting with `pre1` on the left, then `not4`, both timers `tim` and `tim2` and both greater equal threshold blocks, as well as `and2` such taht `mulOr` output would go straight to the latch. I'd give it a try! Now you have all the validation results written, so if you run the test with the new version you'll know right away if anything has changed.

Reply:
I tried this and it seems to be working fine. Submitting the PR for re-review